clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/CacheController.groovy

```diff
@@ -29,22 +30,12 @@ class CacheController {
   @Autowired
   List<OnDemandCacheUpdater> onDemandCacheUpdaters
 
-  @Deprecated
-  @RequestMapping(method = RequestMethod.POST, value = "/{type}")
-  ResponseEntity handleOnDemand(@PathVariable String type, @RequestBody Map<String, ? extends Object> data) {
-    for (updater in onDemandCacheUpdaters) {
-      if (updater.handles(type)) {
-        updater.handle(type, data)
-      }
-    }
-    new ResponseEntity(HttpStatus.ACCEPTED)
-  }
-
   @RequestMapping(method = RequestMethod.POST, value = "/{cloudProvider}/{type}")
   ResponseEntity handleOnDemand(@PathVariable String cloudProvider,
                                 @PathVariable String type,
                                 @RequestBody Map<String, ? extends Object> data) {
-    def cacheStatus = onDemandCacheUpdaters.find { it.handles(type, cloudProvider) }?.handle(type, cloudProvider, data)
+    OnDemandAgent.OnDemandType onDemandType = OnDemandAgent.OnDemandType.fromString(type)
```

Comment:
Is there any known situation where we're sending an invalid type across from orca?

With this throwing a 404 it'll probably now result in the task/pipeline failing.